docs/reference/api/c-api.md

```diff
@@ -66,20 +66,32 @@ is as follows
+         * ```initial_chunk_size_bytes```: This config is only relevant when the arena extend strategy is kNextPowerOfTwo. This is the (possible) size of the region that the arena allocates in the very first allocation (if the first memory request is greater than this the allocation size will be larger than this provided value). Chunks are handed over to allocation requests from this region. If the logs show that the arena is getting extended a lot more than expected, you're better off choosing a big enough initial size for this.
+         * ```initial_growth_chunk_size_bytes```: Please read the section ```Cap memory growth of arenas (arena shrinkage)``` prior to reading this section. This config is only relevant when the arena extend strategy is kNextPowerOfTwo. Currently, this value is the (possible) size of the first allocation post an arena shrinkage (a memory request greater than this value post arena shrinkage will result in higher allocation size). The (possible) first allocation by an arena is defined by ```initial_chunk_size_bytes``` and the possible subsequent allocations are ```initial_chunk_size_bytes * 2```, ```initial_chunk_size_bytes * 4```, and so on. If the arena were to shrink (i.e.) de-allocate any of these memory regions, we want to "reset" the size of the first allocation post shrinkage. This is the current definition. In future, this config may be used to control other "growth-centric" actions of the arena (i.e.) the second allocation made by the arena (arena growth) post the very first allocation (initial chunk) may be defined by this parameter.
          * ```max_dead_bytes_per_chunk```: This controls whether a chunk is split to service an allocation request. Currently if the difference between the chunk size and requested size is less than this value, the chunk is not split. This has the potential to waste memory by keeping a part of the chunk unused (hence called dead bytes) throughout the process thereby increasing the memory usage (until this chunk is returned to the arena).
-* **Share initializer(s) between sessions:**
-   * *Description*: This feature allows a user to share the same instance of an initializer across
-+multiple sessions.
+* **Cap memory growth of arenas (arena shrinkage):**
+   * *Description*: One of the core reasons behind using an arena based allocator is to minimize the number of system calls to trigger an allocation by potentially allocating more memory than what is immediately needed and any extra memory allocated is used to serve potential future memory requests without making an additional system call at that point. Currently, at no point is unused memory in the arena freed. This feature allows users to free unused memory ("shrink" the arena) at some cadence. Currently, the only supported cadence is at the end of every Run() (i.e.) if this feature is used, the arena memory is scanned to potentially free up unused memory at the end of every Run().
```

Comment:
>minimize the number of system calls to trigger an allocation by potentially allocating more memory [](http://example.com/codeflow?start=88&length=98)

this is confusing as you're mixing in aspects of the 'normal' behavior (allocate larger chunks to minimize system allocations and fragmentation) with 'exact size' behavior ('potentially' allocating more memory). 

Also this doesn't really 'cap' the growth. If I send in a massive batch the arena won't limit how big it gets during the request. 

As such, maybe just talk about how the arena doesn't shrink by default, and now you can explicitly make it do so at the end of a Run call. It's a run option right, so not 'every' Run - up to the user to choose.

Reply:
Fixed